src/main/java/frc/robot/subsystems/drive/states/LinearDriveState.java

```diff
@@ -0,0 +1,266 @@
+            : new Pose2d(
+                JsonConstants.blueFieldLocations.blueReefOTF11Translation,
+                JsonConstants.blueFieldLocations.blueReefOTF11Rotation);
+      case CoralStationRight:
+        return driveInput.isAllianceRed()
+            ? new Pose2d(
+                JsonConstants.redFieldLocations.redCoralStationRightTranslation,
+                JsonConstants.redFieldLocations.redCoralStationRightRotation)
+            : new Pose2d(
+                JsonConstants.blueFieldLocations.blueCoralStationRightTranslation,
+                JsonConstants.blueFieldLocations.blueCoralStationRightRotation);
+      case CoralStationLeft:
+        return driveInput.isAllianceRed()
+            ? new Pose2d(
+                JsonConstants.redFieldLocations.redCoralStationLeftTranslation,
+                JsonConstants.redFieldLocations.redCoralStationLeftRotation)
+            : new Pose2d(
+                JsonConstants.blueFieldLocations.blueCoralStationLeftTranslation,
+                JsonConstants.blueFieldLocations.blueCoralStationLeftRotation);
+      default:
+        return null;
+    }
+  }
+
+  @Override
+  public void periodic() {
+    Pose2d currentPose = drive.getPose();
+
+    double currentDistance = currentPose.getTranslation().getDistance(goalPose.getTranslation());
+
+    if (withinRange(currentPose, goalPose, 0.05)) {
+      drive.setGoalSpeeds(new ChassisSpeeds(), true);
+
+      if (drive.isDesiredLocationReef()) {
+        drive.fireTrigger(DriveTrigger.BeginLineup);
+      } else {
+        drive.fireTrigger(DriveTrigger.CancelLinear);
+      }
+      return;
+    }
+
+    // DRIVE
+    double ffScaler =
+        MathUtil.clamp(
+            (currentDistance - m_ffMinRadius) / (m_ffMaxRadius - m_ffMinRadius), 0.0, 1.0);
+    double driveVelocityScalar =
+        driveController.getSetpoint().velocity * ffScaler
+            + driveController.calculate(currentDistance, 0.0);
+    if (currentDistance < driveController.getPositionTolerance()) {
+      driveVelocityScalar = 0.0;
+    }
+
+    // HEADING
+    double headingError = currentPose.getRotation().minus(goalPose.getRotation()).getRadians();
+    double headingVelocity =
+        headingController.calculate(
+            currentPose.getRotation().getRadians(), goalPose.getRotation().getRadians());
+    if (Math.abs(headingError) < headingController.getPositionTolerance()) {
+      headingVelocity = 0.0;
+    }
+
+    Translation2d driveVelocity =
+        new Pose2d(
+                0.0, 0.0, currentPose.getTranslation().minus(goalPose.getTranslation()).getAngle())
+            .transformBy(new Transform2d(driveVelocityScalar, 0.0, new Rotation2d()))
+            .getTranslation();
+    ChassisSpeeds speeds =
+        ChassisSpeeds.fromFieldRelativeSpeeds(
+            driveVelocity.getX(), driveVelocity.getY(), headingVelocity, currentPose.getRotation());
+    drive.setGoalSpeeds(speeds, true);
```

Comment:
Shouldn't this be passing false? I think we're converting from field relative to robot relative speeds above.